hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -402,9 +405,32 @@ private void setHmsTableParameters(String newMetadataLocation, Table tbl, TableM
       parameters.put(StatsSetupConst.TOTAL_SIZE, summary.get(SnapshotSummary.TOTAL_FILE_SIZE_PROP));
     }
 
+    setSnapshotStats(metadata, parameters);
+
     tbl.setParameters(parameters);
   }
 
+  private void setSnapshotStats(TableMetadata metadata, Map<String, String> parameters) {
+    Snapshot currentSnapshot = metadata.currentSnapshot();
```

Comment:
I think we need a way to clear the properties if they are not there anymore in current snapshot (like original code setHmsProperties).  For example if the summary becomes suddenly over the length in new snapshot, then it will not override the old summary and would be wrong.

Reply:
Fixed in the new commit.